lib/js/controllers/s-popover.js

```diff
@@ -0,0 +1,94 @@
+(function () {
+    "use strict";
+    Stacks.addController("s-popover", {
+        targets: [],
+        
+        /**
+         * Initializes popper.js and document events on controller connect
+         */
+        connect: function() {
+            var popoverId = this.element.getAttribute("aria-controls");
+
+            if (!popoverId) {
+                throw "[aria-controls=\"{POPOVER_ID}\"] required";
+            }
+
+            this.popoverElement = document.getElementById(popoverId);
+
+            this.popper = new Popper(this.element, this.popoverElement, {
+                placement: this.data.get("placement") || "bottom",
+            });
+
+            // in order for removeEventListener to remove the right event, this bound function needs a constant reference
+            this._boundClickFn = this._hideOnOutsideClick.bind(this);
+
+            document.addEventListener("click", this._boundClickFn);
+        },
+
+        /**
+         * Cleans up popper.js elements and disconnects all added event listeners on controller disconnect
+         */
+        disconnect: function() {
+            this.popper.destroy();
+
+            document.removeEventListener("click", this._boundClickFn);
+        },
+
+        /**
+         * Toggles the visibility of the popover when called as a Stimulus action
+         * @param {Event} event - The event object from the Stimulus action call
+         */
+        toggle: function(event) {
+            this._toggle();
+        },
+
+        /**
+         * Shows the popover
+         */
+        show: function() {
+            this._toggle(true);
+        },
+
+        /**
+         * Hides the popover
+         */
+        hide: function() {
+            this._toggle(false);
+        },
+
+        /**
+         * Toggles the visibility of the popover element
+         * @param {boolean=} show - Optional parameter that force shows/hides the element or toggles it if left undefined
+         */
+        _toggle: function(show) {
+            this.popoverElement.classList.toggle("is-visible", show);
+            this._toggleOptionalClasses(show);
+        },
+
+        /**
+         * Forces the popover to hide if a user clicks outside of it or its reference element
+         * @param {Event} e - The document click event
+         */
+        _hideOnOutsideClick: function(e) {
```

Comment:
We should also handle `ESC` keys, returning focus to the source element if focus was within the popover.  Here's my code for that:

    var possibleChildSelector = '.s-modal:not([aria-hidden=false]), .s-toast:not([aria-hidden=false])';

        documentKeyup: function (e) {
            var target = e.target;
            if (e.which === 27 && this.isVisible && $(possibleChildSelector).length === 0) {
                if (this.$popover.has(target).length > 0 || this.$popover.is(target)) { this.$source.focus(); }
                this._toggle();
            }
        },